ndk-build/src/manifest.rs

```diff
@@ -258,6 +262,39 @@ pub struct Permission {
     pub max_sdk_version: Option<u32>,
 }
 
+/// Android [package element](https://developer.android.com/guide/topics/manifest/queries-element#package).
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct Package {
+    #[serde(rename(serialize = "android:name"))]
+    pub name: String,
+}
+
+/// Android [provider element](https://developer.android.com/guide/topics/manifest/queries-element#provider).
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct QueryProvider {
+    #[serde(rename(serialize = "android:authorities"))]
+    pub authorities: String,
+
+    // The specs say only `authorities` attribute is the only one needed for providers contained in `queries`
+    // however aapt throws errors requireing a `android:name` attribute, nd-build should be using appt2?
+    #[serde(rename(serialize = "android:name"))]
+    pub name: String,
+}
+
+/// Android [queries element](https://developer.android.com/guide/topics/manifest/queries-element)
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct Query {
```

Comment:
I think this should be named `Queries` as it is hinting at being a verb, not a plural list of "multiple queries".